apps/wallet-mobile/src/features/Receive/common/ShareQRCodeCard/ShareQRCodeCard.tsx

```diff
@@ -0,0 +1,197 @@
+import {useTheme} from '@yoroi/theme'
+import React, {RefObject, useEffect, useRef, useState} from 'react'
+import {StyleSheet, TouchableOpacity, useWindowDimensions, View} from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import QRCode from 'react-native-qrcode-svg'
+import Animated, {FadeInDown, FadeOutDown, Layout} from 'react-native-reanimated'
+import Share from 'react-native-share'
+import ViewShot, {captureRef} from 'react-native-view-shot'
+
+import {Text} from '../../../../components'
+import {CaptureShareQRCodeCard} from '../CaptureShareQRCodeCard/CaptureShareQRCodeCard'
+import {mocks} from '../mocks'
+import {useStrings} from '../useStrings'
+
+type ShareProps = {
+  address?: string
+  title?: string
+  addressDetails?: AddressDetailsProps
+  isCopying?: boolean
+  onLongPress?: () => void
+}
+
+type AddressDetailsProps = {
+  address: string
+  stakingHash?: string
+  spendingHash?: string
+  title?: string
+}
+
+export function ShareQRCodeCard({address, title, isCopying, onLongPress}: ShareProps) {
+  const [isSharing, setIsSharing] = useState<boolean>(false)
+  const strings = useStrings()
+  const ref: RefObject<ViewShot> = useRef(null)
+
+  const {styles, colors} = useStyles()
+
+  const shareImage = () => {
+    setIsSharing(true)
+  }
+
+  useEffect(() => {
+    if (isSharing) {
+      const captureAndShare = async () => {
+        try {
+          await new Promise((resolve) => setTimeout(resolve, 50))
+
+          const uri = await captureRef(ref, {
+            format: 'png',
+            quality: 1,
+            fileName: mocks.shareFileName,
+          })
+
+          setIsSharing(false)
+          await Share.open({url: uri, filename: mocks.shareFileName, message: `${strings.address} ${address}`})
+        } catch (error) {
+          setIsSharing(false)
```

Comment:
avoid using errors as logic
finally -> setIsSharing(false), what would be displayed to the user if the logic eventually fails? is there an error state on design for it?

Reply:
There is no design for error.